src/file_pipe_log/pipe_builder.rs

```diff
@@ -80,9 +81,14 @@ pub struct DualPipesBuilder<F: FileSystem> {
     /// Only filled after a successful call of `DualPipesBuilder::scan`.
     dirs: Paths,
     dir_locks: Vec<StdFile>,
+
     append_files: Vec<File<F>>,
     rewrite_files: Vec<File<F>>,
     recycled_files: Vec<File<F>>,
+
+    pub(crate) append_file_names: Vec<FileName>,
```

Comment:
Move these between dir_locks and append_files, the same order as they are created.